### ISSUE TITLE
netlab connect can't handle `ansible_connection: httpapi`

### ISSUE THREAD

**Issue by petercrocker** [ipspace/netsim-tools#153]
Fortinet is defined in topology-defaults.yml with:

```yaml
  ansible_connection: httpapi
```

Result is:
```
~/forti-temp:    netlab connect fw
Fatal error in connect: Unknown connection method httpapi for host fw
```

Problem is that gets passed to `netlab connect`. What's needed is to use `httpapi` when attempting to configure the device, but `ssh` is used by `netlab connect`. Not sure the best way to fix this without breaking other vendors, hence opening a bug for input.

**Comment by jbemmel** [comment]
I had a similar issue with gNMI: https://github.com/ipspace/netsim-tools/blob/master/netsim/ansible/tasks/deploy-config/sros.yml#L89

My solution was to use the default ansible_connection, and only override it for the tasks that use gNMI

**Comment by ipspace** [comment]
@petercrocker: thanks for the bug report
@jbemmel: next time please tell me what's wrong and I'll fix it ;)

There are two ways to fix this:

* Use **ssh** (plus a warning) whenever an unknown connection type is encountered
* Use another Ansible group variable (maybe **netsim_console_connection**) to override **ansible_connection** when needed.

Which one would you prefer?

**Comment by ipspace** [comment]
Don't worry, I'll fix **netlab connect**. Please leave your tasks as they are, we don't want to create unnecessary confusion (as you wrote, everyone working on that bit of code would have to figure out what we did).

**Comment by petercrocker** [comment]
Ok thanks @ipspace i'll hold off trying any more workarounds around this.

**Comment by ipspace** [comment]
Fixed in 1bf74d7. 

@jbemmel: You could simplify your Ansible task lists and device settings if you wish. I added **netsim_console_connection** to **sros** and **srlinux**, hope that's OK.

**Comment by jbemmel** [comment]
One can use a “block:” and a block variable for ansible_connection to set it for a number of tasks

Like ‘ansible_connection’, netsim uses ‘ansible_network_os’ to drive its logic, it would be better to replace that with ‘netsim_network_os’ too (as it’s unrelated to how Ansible uses that variable)
